### PR TITLE
old PR

### PR DESCRIPTION
**Pull request by sevu**

### CODE REVIEW

File: units/spelk_soulless.cfg [hunk 1]
```diff
@@ -37,49 +37,15 @@
...
-        [frame]
-            duration=150
-            image="units/undead/soulless-die-10.png"
+            image="units/undead/soulless-die-[5~10].png:150"
         [/frame]
     [/death]
     [attack]
         name=touch
-        description=_"touch"
+        description={STR_TOUCH}
```
Comment (konecnyjakub) [inline units/spelk_soulless.cfg:48]: This is not necessary as we are in wesnoth-units textdomain.

Comment (sevu) [reply units/spelk_soulless.cfg:48]: Oh, I didn't even see. But I would change it anyway.

**Review by konecnyjakub — COMMENTED** [review]
Wow, this is a big clean up. Using the new format for animations is a good thing as is basing new units on already existing ones. Using images from mainline is also nice.

File: units/L_Ghast.cfg [hunk 2]
```diff
@@ -2,41 +2,20 @@
...
-        name=bite
-        description={STR_BITE}
-        icon=attacks/fangs.png
-        type=blade
-        range=melee
-        damage=10
-        number=3
-        [specials]
-            {WEAPON_SPECIAL_POISONG}
-        [/specials]
```
Comment (konecnyjakub) [inline units/L_Ghast.cfg:38]: Well, we actually want to change that attack. The unit type description says that the poison works also on the undead and the poisong weapon special does that (via events).

Comment (sevu) [reply units/L_Ghast.cfg:38]: If it worked before it should still work. Actually nothing changed, the primary attack has already the weapon special from the base_unit. Having the empty [attack] block means that the second [attack] block is not changing the first attack which we did inherit, but adding a new one.

**Comment by sevu** [comment]
Will do that for some more units too.
What I just saw: the custom ghast (the one without L) is used nowhere at all. Maybe you find a use for it.

File: units/L_Ghast.cfg (hunk 2 above)
Comment (sevu) [reply units/L_Ghast.cfg:38]: Maybe you had the problem which the next commit avoids.

Comment (konecnyjakub) [reply units/L_Ghast.cfg:38]: No. The base unit has {WEAPON_SPECIAL_POISON} which comes from core and does not work on undead. {WEAPON_SPECIAL_POISONG} is defined by this campaign and is an extended version that also work on undead. The events for it are in utils/abilities_events.cfg in macro POISONG.

Comment (sevu) [reply units/L_Ghast.cfg:38]: ops, I missed that.

Comment (sevu) [reply units/L_Ghast.cfg:38]: Seems I can't use [base_unit] then.

Comment (konecnyjakub) [reply units/L_Ghast.cfg:38]: Ah, yes. Just tested it and the unit had both regular poison and the custom one. Actually I have already seen unit type taking weapon special from base unit. Dunno whatever it is a bug in Wesnoth or a feature ...

Comment (sevu) [reply units/L_Ghast.cfg:38]: you talk about the case that the code looks like this?
	
[unit_type]
    id=L Ghast

	[base_unit]
	    id=Ghast
	[/base_unit]

	[attack]
        [specials]
            {WEAPON_SPECIAL_OUR_MODIFIED_SPECIAL}
        [/specials]
	[/attack]
[/unit_type]

I think the problem why it doesn't overwrite the original weapon special is that the tag names are different,  [poision_g] while the original is [poison].

Comment (konecnyjakub) [reply units/L_Ghast.cfg:38]: What I meant is that here https://github.com/konecnyjakub/For_Power/blob/54c43c3ebd498ac7c3125fb404ed708fc630407d/units/humans/cult.cfg#L13 the first attack inherits weapon special from the base unit.

Comment (sevu) [reply units/L_Ghast.cfg:38]: The problem is, you can't remove things with [base_unit].
The weapon special macros are in the end all different tags. You could change things in the tag, e.g. the % in [chance_to_hit], but not the whole tag.

In your case

[attack]
[/attack]

[attack]
[/attack]

  [attack]
    name=staff
    description={STR_STAFF}
    icon=attacks/staff-ruby.png
    type=impact
    range=melee
    damage=5
    number=2
  [/attack]

would work, but have the attack at the end.

This isn't quite ideal, I would like to propose a change to [base_unit], but I don't really know how it should be.

**Comment by konecnyjakub** [comment]
Uhm, why did you close this? Because of the merge conflicts? You could resolve them with rebase + force push.

**Comment by sevu** [comment]
just did that. I thought I make it as a new PR

**Pull request closed without merging**